bayesian_inversion.m

```diff
@@ -0,0 +1,211 @@
+%% RUN A BAYESIAN INVERSION  
+%
+%  Here we will solve for the best T, phi, g for a given 
+%  tomography model using non-linear Bayesian inference
+
+
+function bayesian_inversion(observations, vbr_predictions)
+    
+    addpath('./functions')
+    addpath('./inv_functions')
+
+    % extract variables from observations structure
+    vs = observations.vs;
+    Q = observations.Q;
+    lons = observations.lons;
+    lats = observations.lats;
+    zs = observations.zs;
+    nlat = observations.nlat;
+    nlon = observations.nlon;
+    nz = observations.nz;
+    npts = observations.npts;
+    vs_err = observations.vs_err;
+    Q_err = observations.Q_err;
+    vs_vbr = vbr_predictions.vs_vbr;
+    Q_vbr = vbr_predictions.Q_vbr;
+    phi = vbr_predictions.phi;
+    g = vbr_predictions.g;
+    T = vbr_predictions.T;
+    nT = length(T);
+    ng = length(g);
+    nphi = length(phi);
+    
+    
+    % calculate log10(Q) quantities
+    
+    lQ = log10(Q);
+    lQ_err = log10(Q_err);
+    lQ_vbr= log10(Q_vbr);
+
+    %% BEGIN MAKING INITIAL VECTORS AND MATRICES
+
+    %  For all spatial dimensions, the loop order is same as tomography:
+    %      for ilat=1:nlat ... Latitude
+    %         for ilon=1:nlon ... Longitude
+    %            for iz=1:nz ... Depth
+    %  The order for the model parameters is T1, phi1, g1, T2, phi2, g2, ...
+    %  Our total model space is (npts * number of physical parameters):
+    npar = 3;
+    pars = ['T','phi','g'];
+    nmod = npts*npar;
+    ntype = 2; % type of data (vs and Q)
+    ndata = npts*ntype; % total data
+
+    %  (1) Starting (X0) and Prior (Xpr) Models
+    %  Let's assume X0 and Xpr are the same.
+    phi0 = 0.01; % melt fraction
+    g0 = log10(1000.); % 1 mm (1000 microns)
+    % g0 = 4000.; % 5 mm
+
+    [X0,iT,iphi,ig] = make_X0_xfit_premelt(vs,nz,nlat,nlon,nmod,npts,vs_vbr,phi0,g0,phi,g,T);
```

Comment:
swap "make_X0_xfit_premelt" for generalised "make_X0"

Reply:
added a TO DO comment as I wasn't sure the `make_X0` on my branch would work as is -- we need to pass it a q_method string I think.

Reply:
oh, oops. just reealized that vs here is already selected for anelastic method, so just switching to `make_X0` should be fine now. In any case, I'll leave the TO DO for now and leave it to a subsequent PR...